### PR TITLE
Yaml library update

### PR DESCRIPTION
**Pull request by Mike7311**
Changed library for working with YAML from JYaml to Jackson Yaml

Removed the JYaml dependency:
```
<groupId>org.jyaml</groupId>
<artifactId>jyaml</artifactId>
```
And added Jackson YAML:
```
<groupId>com.fasterxml.jackson.dataformat</groupId>
<artifactId>jackson-dataformat-yaml</artifactId>
```

Due to the new library being stricter on the format of the Yaml, some changes in the configuration files were made, to make them compliant with yaml format:
- removed not allowed characters from property values (@ in @shared.privileged.geoserver.pass@)
- removed references without anchors (document declares link to *commonFooter but anchor &commonFooter was not defined previously)
- added a space between key\value pairs ("absoluteX: 30" instead of "absoluteX:30")

Some additional issues were discovered when trying to parse this [configuration file](https://github.com/geosolutions-it/MapStore2/blob/master/java/printing/resources/geoserver/print/config.yaml):

```
- !ipMatch
    host: 127.0.0.1
    # Allow to all hosts
    mask: 0.0.0.0
```
!ipMatch is mapped to [AddressHostMatcher](https://github.com/geosolutions-it/mapfish-print-v2/blob/0c055e2a36bec3b12eafd207144ff8eb7b37f987/src/main/java/org/mapfish/print/config/AddressHostMatcher.java#L26), which does not have host property. Should instead use ip property (e.g ip: 127.0.0.1)
```
- !text
              width: 300
              text: '${comment}'
              ...
```
!text is mapped to [TextBlock class](https://github.com/geosolutions-it/mapfish-print-v2/blob/4b73912e5565ae206af5b5c434ef37454a98c252/src/main/java/org/mapfish/print/config/layout/TextBlock.java#L37). It does not have width property, so it should be removed.
```
lastPage:
    rotation: true
    ...
```
lastPage is mapped to [LastPage class](https://github.com/mapfish/mapfish-print-v2/blob/4b73912e5565ae206af5b5c434ef37454a98c252/src/main/java/org/mapfish/print/config/layout/LastPage.java#L30), and does not have rotation property.

Attaching config file updated with the mentioned fixes:

[config.zip](https://github.com/geosolutions-it/mapfish-print-v2/files/14061761/config.zip)

This library version was tested with mapstore.war downloaded from [HERE](https://github.com/geosolutions-it/MapStore2/releases/tag/v2023.02.01) and deployed on tomcat 9.0.74, with instructions provided [HERE](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/printing-module/#adding-to-an-existing-mapstore).
Also, the following jars were added to WEB-INF/lib folder:
```
json-20230618.jar
openpdf-1.3.26.jar
jackson-dataformat-yaml-2.14.0.jar
snakeyaml-1.33.jar
```
These jars were taken from printing/target/mapstore-printing/WEB-INF/lib of mapstore.

Also attaching print made with this version of library
[mapstore-print.pdf](https://github.com/geosolutions-it/mapfish-print-v2/files/14061900/mapstore-print.pdf)

### CODE REVIEW

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/config-georchestra.yaml [hunk 1]
```diff
@@ -48,9 +48,9 @@ brokenUrlPlaceholder: http://upload.wikimedia.org/wikipedia/commons/c/ce/Transpa
 security:
   - !basicAuth
     matcher: !dnsMatch
-      host: @shared.server.name@
-    username: @shared.privileged.geoserver.user@
-    password: @shared.privileged.geoserver.pass@
+      host: shared.server.name@
+    username: shared.privileged.geoserver.user
+    password: shared.privileged.geoserver.pass
```
Comment (afabiani) [inline src/test/resources/org/mapfish/print/config/sample_config_yaml/config-georchestra.yaml:53]: Why we removed the `@` characters here? Is there a specific reason?

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configNonPremiumGoogle.yaml [hunk 2]
```diff
@@ -44,5 +44,4 @@ layouts:
         - !map
           spacingAfter: 30
           width: 440
-          height: 483
-      footer: *commonFooter
```
Comment (afabiani) [inline src/test/resources/org/mapfish/print/config/sample_config_yaml/configNonPremiumGoogle.yaml:48]: Why we removed the `footer` here?

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configPremiumGoogle.yaml [hunk 3]
```diff
@@ -49,5 +49,4 @@ layouts:
         - !map
           spacingAfter: 30
           width: 440
-          height: 483
-      footer: *commonFooter
```
Comment (afabiani) [inline src/test/resources/org/mapfish/print/config/sample_config_yaml/configPremiumGoogle.yaml:53]: Same as above, Why we removed the footer here?

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configWMTS.yaml [hunk 4]
```diff
@@ -45,4 +45,3 @@ layouts:
           spacingAfter: 30
           width: 440
           height: 483
-      footer: *commonFooter
```
Comment (afabiani) [inline src/test/resources/org/mapfish/print/config/sample_config_yaml/configWMTS.yaml:48]: Again, Why we removed the footer here?

**Review by afabiani — CHANGES_REQUESTED** [review]
More or less is good, but I see that we removed some `EnumWrappers` and the test case for it. So, I'm wondering if the new version will be able to parse correctly the same `YAML` as before. We should add a test case trying to parse a sampe `YAML` file and verify we are able to fetch correctly all its parts.

**Review by tdipisa — COMMENTED** [review]
@Mike7311 @afabiani @nmco 
just remember that the [mapfish-print-v2](https://github.com/geosolutions-it/mapfish-print-v2) is currently the reference also for the gs-printing. Therefore we should also take into account this in the review and not only MS (as I reported [here](https://github.com/geosolutions-it/support/issues/2992#issuecomment-1889340528)). I hope you have discussed this before proceeding with this PR.

Anyway MS is currently using this YAML: https://github.com/geosolutions-it/MapStore2/blob/master/java/printing/resources/geoserver/print/config.yaml

If you want to test it locally you can download the needed MS binary assets [here](https://github.com/geosolutions-it/MapStore2/releases/tag/v2023.02.01) and follow the guide [here](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/printing-module/#adding-to-an-existing-mapstore). **Of course the printing JARs in the final binary installation need to be replaced with the new ones.**

It would be good to understand what need to be changed there to be able to properly document breaking changes in the MS migration guidelines.

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/config-georchestra.yaml (hunk 1 above)
Comment (Mike7311) [reply src/test/resources/org/mapfish/print/config/sample_config_yaml/config-georchestra.yaml:53]: Yes, YAML does not allow that. This is one of the examples of the malformed yaml, where JYaml would simply ignore invalid properties, but the new library throws an exception

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configNonPremiumGoogle.yaml (hunk 2 above)
Comment (Mike7311) [reply src/test/resources/org/mapfish/print/config/sample_config_yaml/configNonPremiumGoogle.yaml:48]: Also a malformed yaml. The * character is [an anchor](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/), which is used to reference previously defined with & character property (for example &commonFooter), but in this file there are no such definition, which results in exception (for correct example of & and * usage you can see [configMultipleMaps.yaml](https://github.com/geosolutions-it/mapfish-print-v2/blob/d91e04ea89dde4015a8675de9648801d8727d623/src/test/resources/org/mapfish/print/config/sample_config_yaml/configMultipleMaps.yaml#L101)).

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configPremiumGoogle.yaml (hunk 3 above)
Comment (Mike7311) [reply src/test/resources/org/mapfish/print/config/sample_config_yaml/configPremiumGoogle.yaml:53]: Same as above answer

File: src/test/resources/org/mapfish/print/config/sample_config_yaml/configWMTS.yaml (hunk 4 above)
Comment (Mike7311) [reply src/test/resources/org/mapfish/print/config/sample_config_yaml/configWMTS.yaml:48]: same as [here](https://github.com/geosolutions-it/mapfish-print-v2/pull/3#discussion_r1464358788)

**Comment by Mike7311** [comment]
@afabiani. There are some tests for checking yaml parsing in the project. For example [ConfigTest](https://github.com/geosolutions-it/mapfish-print-v2/blob/573a75dc581b905bfc473ca13e4f10cafc10570b/src/test/java/org/mapfish/print/config/ConfigTest.java#L42) and [PdfUtilsTest](https://github.com/geosolutions-it/mapfish-print-v2/blob/4b73912e5565ae206af5b5c434ef37454a98c252/src/test/java/org/mapfish/print/PDFUtilsTest.java#L44). All tests were green with changes in this PR.

**Review by afabiani — APPROVED** [review]

**Pull request merged**